python/lsst/ts/externalscripts/auxtel/calsys_take_narrowband_data.py

```diff
@@ -311,23 +404,26 @@ def set_metadata(self, metadata):
         await asyncio.sleep(self.fiber_spectrometer_delays[index])
 
         timeout = self.integration_times[index] + self.cmd_timeout
-        fiber_spectrograph_lfo_coro = self.fiber_spectrograph.evt_largeFileObjectAvailable.next(timeout=self.cmd_timeout,flush=True)
+        fiber_spectrograph_lfo_coro = self.fiber_spectrograph.evt_largeFileObjectAvailable.next(
+            timeout=self.cmd_timeout, flush=True)
         self.fiber_spectrograph.cmd_captureSpectImage.set(
             imageType=self.image_types[index],
             integrationTime=self.fiber_spectrograph_integration_times[index],
-            lamp=self.lamps[index],
+            lamp=self.lamps[index]
         )
         self.log.info(f"take a {self.integration_times[index]} second exposure")
         await self.fiber_spectrograph.cmd_captureSpectImage.start(timeout=timeout)
         self.log.debug(f"Fiber Spectrograph captured spectrum image")
         return await fiber_spectrograph_lfo_coro
-    
+
     async def start_camera_take_image(self, index):
-        self.atcamera.cmd_takeImages.set(shutter=self.shutter[index],
-                                                 numImages=1,
-                                                 expTime=self.integration_times[index],
-                                                 imageSequenceName=self.image_sequence_name[index])
-        atarchiver_lfo_coro = self.atarchiver.evt_processingStatus.next(flush=True,timeout=(self.cmd_timeout*2)+self.integration_times[index])
+        self.atcamera.cmd_takeImages.set(
+            shutter=self.shutter[index],
+            numImages=1,
+            expTime=self.integration_times[index],
+            imageSequenceName=self.image_sequence_name[index])
+        atarchiver_lfo_coro = self.atarchiver.evt_processingStatus.next(flush=True, timeout=(
+            self.cmd_timeout*2)+self.integration_times[index])
```

Comment:
I suspect this (pre-existing) code code not do what you think it does. The flush in evt_processingStatus.next will not occur until you start awaiting it. I think what you want is as follows (I also combined the set and start, to save a line, but explicitly calling flush first is the important change):

    self.atarchiver.evt_processingStatus.flush()
    await self.atcamera.cmd_take_images.set_start(
       shutter=...
    )
    return self.atarchiver.evt_processingStatus.next(flush=True, timeout=...)